From 4e05cc26507715c619c5677bd76557ddc15826b7 Mon Sep 17 00:00:00 2001
From: Volodymyr Bendiuga <volodymyr.bendiuga@westermo.se>
Date: Fri, 27 May 2016 15:29:17 +0200
Subject: [PATCH 22/38] route/link: set NLA_F_NESTED flag when creating nested
 AF_INET
Organization: Westermo Network Technologies AB

Linux kernel expects this to be set, otherwise warning will be generated.
This flag is needed to help kernel differentiate between interface
attributes and bridge attributes.

Signed-off-by: Volodymyr Bendiuga <volodymyr.bendiuga@westermo.se>
Signed-off-by: Joachim Nilsson <troglobit@gmail.com>
---
 lib/route/link.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/route/link.c b/lib/route/link.c
index a53ff00..06e6d75 100644
--- a/lib/route/link.c
+++ b/lib/route/link.c
@@ -167,7 +167,7 @@ static int af_fill(struct rtnl_link *link, struct rtnl_link_af_ops *ops,
 		return 0;
 
 	if (!ops->ao_fill_af_no_nest)
-		if (!(af_attr = nla_nest_start(msg, ops->ao_family)))
+		if (!(af_attr = nla_nest_start(msg, ops->ao_family | NLA_F_NESTED)))
 			return -NLE_MSGSIZE;
 
 	if ((err = ops->ao_fill_af(link, arg, data)) < 0)
-- 
2.25.1

